plugins/parsers/hep/hep.go

```diff
@@ -0,0 +1,152 @@
+	Protocol    uint32
+	SrcIP       string
+	DstIP       string
+	SrcPort     uint32
+	DstPort     uint32
+	Tsec        uint32
+	Tmsec       uint32
+	ProtoType   uint32
+	NodeID      uint32
+	NodePW      string
+	Payload     string
+	CID         string
+	Vlan        uint32
+	ProtoString string
+	Timestamp   time.Time
+	NodeName    string
+}
+
+func (h *HEP) parseHEP(packet []byte) error {
+	length := binary.BigEndian.Uint16(packet[4:6])
+	if int(length) != len(packet) {
+		return fmt.Errorf("HEP packet length is %d but should be %d", len(packet), length)
+	}
+	currentByte := uint16(6)
+
+	for currentByte < length {
+		hepChunk := packet[currentByte:]
+		if len(hepChunk) < 6 {
+			return fmt.Errorf("HEP chunk must be >= 6 byte long but is %d", len(hepChunk))
+		}
+		//chunkVendorId := binary.BigEndian.Uint16(hepChunk[:2])
+		chunkType := binary.BigEndian.Uint16(hepChunk[2:4])
+		chunkLength := binary.BigEndian.Uint16(hepChunk[4:6])
+		if len(hepChunk) < int(chunkLength) || int(chunkLength) < 6 {
+			return fmt.Errorf("HEP chunk with %d byte < chunkLength %d or chunkLength < 6", len(hepChunk), chunkLength)
+		}
+		chunkBody := hepChunk[6:chunkLength]
+
+		switch chunkType {
+		case Version, Protocol, ProtoType:
+			if len(chunkBody) != 1 {
+				return fmt.Errorf("HEP chunkType %d should be 1 byte long but is %d", chunkType, len(chunkBody))
+			}
+		case SrcPort, DstPort, Vlan:
+			if len(chunkBody) != 2 {
+				return fmt.Errorf("HEP chunkType %d should be 2 byte long but is %d", chunkType, len(chunkBody))
+			}
+		case IP4SrcIP, IP4DstIP, Tsec, Tmsec, NodeID:
+			if len(chunkBody) != 4 {
+				return fmt.Errorf("HEP chunkType %d should be 4 byte long but is %d", chunkType, len(chunkBody))
+			}
+		case IP6SrcIP, IP6DstIP:
+			if len(chunkBody) != 16 {
+				return fmt.Errorf("HEP chunkType %d should be 16 byte long but is %d", chunkType, len(chunkBody))
+			}
+		}
+
+		switch chunkType {
+		case Version:
+			h.Version = uint32(chunkBody[0])
+		case Protocol:
+			h.Protocol = uint32(chunkBody[0])
+		case IP4SrcIP:
+			h.SrcIP = net.IP(chunkBody).To4().String()
+		case IP4DstIP:
+			h.DstIP = net.IP(chunkBody).To4().String()
+		case IP6SrcIP:
+			h.SrcIP = net.IP(chunkBody).To16().String()
+		case IP6DstIP:
+			h.DstIP = net.IP(chunkBody).To16().String()
+		case SrcPort:
+			h.SrcPort = uint32(binary.BigEndian.Uint16(chunkBody))
+		case DstPort:
+			h.DstPort = uint32(binary.BigEndian.Uint16(chunkBody))
+		case Tsec:
+			h.Tsec = binary.BigEndian.Uint32(chunkBody)
+		case Tmsec:
+			h.Tmsec = binary.BigEndian.Uint32(chunkBody)
+		case ProtoType:
+			h.ProtoType = uint32(chunkBody[0])
+			switch h.ProtoType {
+			case 1:
+				h.ProtoString = "sip"
+			case 5:
+				h.ProtoString = "rtcp"
+			case 34:
+				h.ProtoString = "rtpagent"
+			case 35:
+				h.ProtoString = "rtcpxr"
+			case 38:
+				h.ProtoString = "horaclifix"
+			case 53:
+				h.ProtoString = "dns"
+			case 100:
+				h.ProtoString = "log"
+			case 112:
+				h.ProtoString = "alert"
```

Comment:
Can you please _define_ those magic numbers similarly to the chunk-type etc?

Reply:
Protocol strings are the equivalent of a `type` tag and are injected by HEP senders to further identify the payload type.

Reply:
I don't doubt any of your words, but it won't hurt to define those numbers as consts and use them as speaking names. You can the even implement a type with the `Stringer` interface to cover this conversion here.